packages/application-components/src/components/modal-pages/internals/modal-page.styles.ts

```diff
@@ -32,7 +33,11 @@ export const getContainerStyles = (props: StyleProps): SerializedStyles => css`
   right: 0;
   height: 100%;
   width: ${props.size !== 'scale'
-    ? `${stylesBySize[props.size].width} !important`
+    ? // In case we're using a specific size, we want it to be used until there's no space left.
+      // In such scenario, we want the modal to be as wide as possible, but using the shared indentation width size.
+      `min(${
```

Comment:
We use the minimum value between the expected size for the modal and the available space in its container (subtracting the indentation gap we use for modals).